test/functional/forall/atomic/tests/test-forall-atomic-basic.hpp

```diff
@@ -25,10 +27,11 @@ class ForallAtomicBasicFunctionalTest : public ::testing::Test
 template <typename ExecPolicy,
           typename AtomicPolicy,
           typename WORKINGRES,
+          typename SegmentType,
```

Comment:
I would have taken a simpler implementation route (that I described on the confluence page) to reduce the depth of googletest template nesting.If this doesn't take too long to compile and link, it works for me.

Reply:
Yes, I was thinking about the compilation time as well, but it does not seem to add that much compilation time.